### PR TITLE
Add support for Laravel Vapor

### PR DESCRIPTION
**Pull request by d8vjork**
**Changes proposed:**

- [x] Add
- [ ] Delete
- [ ] Fix
- [ ] Prepare

### CODE REVIEW

File: src/iconsManifest/supportedExtensions.ts [hunk 1]
```diff
@@ -3961,6 +3961,7 @@ export const extensions: IFileCollection = {
     { icon: 'vala', extensions: ['vala'], format: FileFormat.svg },
     { icon: 'vapi', extensions: ['vapi'], format: FileFormat.svg },
     { icon: 'vash', extensions: ['vash'], light: true, format: FileFormat.svg },
+    { icon: 'vapor', extensions: ['vapor.yml'], format: FileFormat.svg },
```
Comment (robertohuertasm) [inline src/iconsManifest/supportedExtensions.ts:3964]: Is this intended to be an extension like `a.vapor.yaml`? Or will it be the name of a file? If it's the later, then you should add `filename: true` and that will make it (similar to [this](https://github.com/vscode-icons/vscode-icons/blob/cf315a35e926fb9f225e15406de8316f41ad7c7e/src/iconsManifest/supportedExtensions.ts#L2662-L2672))

**Review by robertohuertasm — CHANGES_REQUESTED** [review]
Thanks for this. Just a minor doubt before merging.

File: src/models/language/languageCollection.ts [hunk 2]
```diff
@@ -225,6 +225,7 @@ export interface ILanguageCollection extends INativeLanguageCollection {
   tuc: ILanguage;
   twig: ILanguage;
   typo3: ILanguage;
+  vapor: ILanguage;
```
Comment (robertohuertasm) [inline src/models/language/languageCollection.ts:228]: Note that language is only needed if you want to add language support. This basically means that if there's an extension in the marketplace supporting vapor, we would use it's configuration in order to decide which extensions to use.

File: src/iconsManifest/languages.ts [hunk 3]
```diff
@@ -286,6 +286,7 @@ export const languages: ILanguageCollection = {
   typescript: { ids: 'typescript', defaultExtension: 'ts' },
   typescriptreact: { ids: 'typescriptreact', defaultExtension: 'tsx' },
   typo3: { ids: 'typoscript', defaultExtension: 'typoscript' },
+  vapor: { ids: 'vapor', defaultExtension: 'vapor.yml' },
```
Comment (robertohuertasm) [inline src/iconsManifest/languages.ts:289]: See my comment about the necessity of using a `language` here. I've searched the marketplace and found out about one extension. The id, though, it's not `vapor` but `vaporlang`. You can check it out [here](https://github.com/minkcv/vaporlang-vscode/blob/master/package.json#L16).

**Review by robertohuertasm — CHANGES_REQUESTED** [review]

File: src/models/language/languageCollection.ts (hunk 2 above)
Comment (d8vjork) [reply src/models/language/languageCollection.ts:228]: Oh ok, then I'll remove it on the next PR, didn't know why was this required, just copy one of the PRs.

File: src/iconsManifest/languages.ts (hunk 3 above)
Comment (d8vjork) [reply src/iconsManifest/languages.ts:289]: No, it's Laravel's Vapor (serverless platform & framework), sorry I didn't describe it on the PR clearly. So it doesn't have an extension (yet), but it'll be cool if it has its own icon as the file is always called vapor.yml.

File: src/iconsManifest/supportedExtensions.ts (hunk 1 above)
Comment (d8vjork) [reply src/iconsManifest/supportedExtensions.ts:3964]: Ok, thanks for the suggestions, changing asap

**Pull request merged**